### DISCUSSION TITLE
Add a {{note_line}} variable

### DISCUSSION THREAD

**Discussion by Sanqui** [Taitava/obsidian-shellcommands#279] [Ideas - done & released]
It would be nice to have a `{{note_line}}` variable (or similar), which would function like `{{note_content}}` except only pass the line the cursor is on.  This would be useful to create commands that operate with a given context (such as a task).  Currently it's possible to do by getting the caret position and cropping the line manually.  That is a lot of busywork, so having it as a built-in feature would make the plugin more versatile.

**Comment by Taitava** [comment]
Hi! Thank you for your suggestion! 🙂 It's a good idea. There's been some discussion about it earlier: https://github.com/Taitava/obsidian-shellcommands/discussions/117

I just haven't decided the variable name for it. I do agree that this variable would be really helpful in many situations! 👍

**Reply by Sanqui** [reply]
Oh, nice!  Sorry, I didn't find that discussion when I tried looking for it.

**Reply by Taitava** [reply]
No problem, there's so many discussions here that I sometimes have trouble finding stuff myself 😄.

**Comment by Taitava** [comment]
# Will be released in `0.17.0`
You can follow the process further in a new issue #282. The name of the variable will be `{{caret_paragraph}}`.

**Reply by Taitava** [reply]
@Sanqui I forgot to mention here that `0.17.0` was released a few days ago. Please let me know if you have any feedback regarding the new variable. 🙂

**Reply by Sanqui** [reply]
Awesome, thank you very much!  I should get to try it out soon!
